### ISSUE TITLE
Using Autonomous Activator to Wrench

### ISSUE THREAD

**Issue by speakerrob** [CoFH/Feedback#458]
I'm trying to have an autonomous activator wrench an item so its orientation changes when I pulse a redstone signal; however, when I apply the shortest redstone pulse that I can (20 ms) to the activator it wrenches the item three times instead of once. Sometimes, it wrenches some other random amount of times, but most of the time it does it thrice. 

I've used the /cofh tps command to confirm my tps is stable at 20.00, with less > 9ms on average. I'm running rr3 Mainline which has Thermal Expansion 4.0.0RC7-141. I'm just curious if this is a bug, I'm not supposed to be using the machine in such a way, etc.

**Comment by zed-hex** [comment]
Hi I’m just some random person testing this bug. The problem you are having is not just with a wrench it is with anything that can be spammed. the best way I found to show this is splash potions the autonomous activator will always use 2 with a quick pulse of Redstone. This looks like it is down to how an autonomous activator functions so I don’t think this is a bug as such. My suggestion on a fix for this problem is add a new Redstone control mode on all machines or just the activator that dose 1 operation for a Redstone pulse since this looks like the easiest  way to fix this problem without braking current functionality of the activator (i might be wrong with easiest).This enhancement could be useful on the other machines in some circumstances as well to greater control over a machine.

**Comment by KingLemming** [comment]
This isn't strictly a bug no, although the shortest redstone pulse you can have is either 50 or 100 ms.

Will see if anything can be done. My gut feeling is that it's actually 100 and that's why you see it 3x.

**Comment by speakerrob** [comment]
@KingLemming yes thank you for the correction, I had twisted my numbers up (got the twenty from the fact that there are twenty 50ms ticks in a game tick). I'm using the pulse former from ProjectRed, and you're right, its output equates to a 100ms game tick (i.e. one redstone tick or two game ticks).

I was just under the impression that the autonomous activator only attempted an operation once a loop, or every 20 ticks. Since this isn't technically a bug, @zed-hex 's mentioning of an augmentation to perform a single operation after receiving a redstone signal would be a much welcomed addition if you have time to implement it in future builds.

**Comment by loordgek** [comment]
yea a redstone pulse is nice to have

**Comment by isochronous** [comment]
Serious necromancy here, but if you use Steve's Factory Manager, its redstone emitter can do down to 1 tick pulses.
